### PR TITLE
RDM-3838 - Gradle migration

### PR DESCRIPTION
**Pull request by gbenadikar**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-3838
https://tools.hmcts.net/jira/browse/RDM-4247

### Change description ###
Migrated performance test project from maven to gradle
Upgraded gatling, spring boot to the latest

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/resources/application.conf [hunk 1]
```diff
@@ -22,23 +22,42 @@ createCaseUrl = "caseworkers/"${userCcdId}"/jurisdictions/PROBATE/case-types/Gra
...
+  userCcdId = 131068
   userImportId = 1
   userProfileUrl = "http://localhost:4453"
   caseDataUrl = "http://localhost:4452"
   caseDefinitionUrl = "http://localhost:4451"
   idamS2SUrl = "http://localhost:4502"
   idamUserTokenUrl = "http://localhost:4501"
-  dataStoreS2STokenGeneratorSecret = "${IDAM_KEY_CCD_DATA_STORE}"
-  gatewayS2STokenGeneratorSecret = "${IDAM_KEY_CCD_GATEWAY}"
+  dataStoreS2STokenGeneratorSecret = "AAAAAAAAAAAAAAAB"
```
Comment (mario-paniccia) [inline src/test/resources/application.conf]: can you put back the Env variables here please?

File: src/test/scala/uk/gov/hmcts/ccd/corecasedata/scenarios/ESStdTest.scala [hunk 2]
```diff
@@ -0,0 +1,208 @@
+package uk.gov.hmcts.ccd.corecasedata.scenarios
+
+import io.gatling.core.Predef._
```
Comment (mario-paniccia) [inline src/test/scala/uk/gov/hmcts/ccd/corecasedata/scenarios/ESStdTest.scala]: how come we are importing these ES test classes as part of this PR? These were supposed to be separately reviewed as part of another PR. If it's not a big work can we leave them out please?

Comment (gbenadikar) [reply src/test/scala/uk/gov/hmcts/ccd/corecasedata/scenarios/ESStdTest.scala]: they were part of ESPT branch that I pulled upon your recommendation 🙂

File: src/test/resources/application.conf (hunk 1 above)
Comment (gbenadikar) [reply src/test/resources/application.conf]: will do

File: src/test/scala/uk/gov/hmcts/ccd/corecasedata/scenarios/ESStdTest.scala (hunk 2 above)
Comment (gbenadikar) [reply src/test/scala/uk/gov/hmcts/ccd/corecasedata/scenarios/ESStdTest.scala]: deleted ES tests, Kapil will raise a separate PR for those

File: src/test/resources/application.conf [hunk 3]
```diff
@@ -75,26 +94,66 @@ cnp_sanbox {
   httpProxyPort = "8080"
   userCcdId = 6687
   userImportId = 445
-  idamS2SUrl = "http://betaDevBccidamS2SLB.reform.hmcts.net"
+  idamS2SUrl = "http://rpe-service-auth-provider-sandbox.service.core-compute-sandbox.internal"
   idamUserTokenUrl = "http://betaDevBccidamAppLB.reform.hmcts.net"
```
Comment (kapiljain786) [inline src/test/resources/application.conf]: This is tactical IDAM.

Can you update this REPO with SIDAM URLs and test it against SIDAM please i.e., SPROD ?

File: src/test/scala/uk/gov/hmcts/ccd/util/CcdTokenGenerator.scala [hunk 4]
```diff
@@ -1,27 +1,83 @@
...
+import org.springframework.http.{HttpEntity, HttpHeaders, MediaType}
 import org.springframework.web.client.RestTemplate
 import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.util.MultiValueMap
 
 object CcdTokenGenerator extends PerformanceTestsConfig with SpringApplicationContext {
 
-  val TOKEN_LEASE_URL = s"$UserAuthUrl/testing-support/lease"
+  val AUTHORISATION_CLIENT_ID = s"ccd_gateway"
```
Comment (mario-paniccia) [inline src/test/scala/uk/gov/hmcts/ccd/util/CcdTokenGenerator.scala]: in this case we can simply have "ccd_gateway"

File: src/test/scala/uk/gov/hmcts/ccd/util/CcdTokenGenerator.scala [hunk 5]
```diff
@@ -1,27 +1,83 @@
...
-    println(s"generated user token $token")
-    "Bearer " + token
+    queryParameters.getOrElse("code", "")
+  }
+
+  private def generateUserAuthToken(authCode: String, redirectUri: String): String = {
+    // retrieve SIDAM authorisation code for user name and password
+    val headers = new HttpHeaders
+    headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED)
+    headers.setAccept(util.Arrays.asList(MediaType.APPLICATION_JSON_UTF8))
```
Comment (mario-paniccia) [inline src/test/scala/uk/gov/hmcts/ccd/util/CcdTokenGenerator.scala]: if you import `import scala.collection.JavaConversions._`
you can just pass here a Scala List and that will be automatically converted
```    
headers.setAccept(List(MediaType.APPLICATION_JSON_UTF8))
```

File: src/test/scala/uk/gov/hmcts/ccd/util/CcdTokenGenerator.scala [hunk 6]
```diff
@@ -36,19 +92,11 @@ object CcdTokenGenerator extends PerformanceTestsConfig with SpringApplicationCo
...
-
-  private def parseJurisdiction(url: String) = {
-    if (!url.contains("jurisdictions")) {
-      None
-    } else {
-      val jurisdiction = url.split("jurisdictions/")(1).split("/")(0)
-      Some(jurisdiction.toLowerCase)
-    }
+  private def parseUrlParameters(url: String) = {
+    url.split("&").map(v => {
```
Comment (mario-paniccia) [inline src/test/scala/uk/gov/hmcts/ccd/util/CcdTokenGenerator.scala]: I assume you want to return the query params as  a Map. 
I'd suggest we use this little library: https://github.com/lemonlabsuk/scala-uri#get-query-string-parameters
it could benefit us in other places as well

Comment (gbenadikar) [reply src/test/scala/uk/gov/hmcts/ccd/util/CcdTokenGenerator.scala]: done

File: src/test/scala/uk/gov/hmcts/ccd/util/CcdTokenGenerator.scala (hunk 5 above)
Comment (gbenadikar) [reply src/test/scala/uk/gov/hmcts/ccd/util/CcdTokenGenerator.scala]: done

File: src/test/scala/uk/gov/hmcts/ccd/util/CcdTokenGenerator.scala (hunk 4 above)
Comment (gbenadikar) [reply src/test/scala/uk/gov/hmcts/ccd/util/CcdTokenGenerator.scala]: done

File: src/test/resources/application.conf (hunk 3 above)
Comment (gbenadikar) [reply src/test/resources/application.conf]: done
